database/coop.py

```diff
@@ -121,6 +121,27 @@ def book(self, tag, values):
         Coop.save_json()
         return ret
 
+    def remove(self, values):
+        ret = ""
+        for val in values:
+            if val in self.leyline:
+                self.leyline.remove(val)
+            elif val in self.weapon:
+                self.weapon.remove(val)
+            elif val in self.talent:
+                self.talent.remove(val)
+            elif val in self.artifact:
+                self.artifact.remove(val)
+            elif val in self.trounce:
+                self.trounce.remove(val)
+            elif val in self.world_boss:
+                self.world_boss.remove(val)
+            else:
+                continue
```

Comment:
Maybe instead of continue, you add onto the return string where val cannot be cancelled.

Reply:
Also if there's a lot of if-else checking, I would personally try applying the open closed principle, but just for you to think about. :)